### PR TITLE
Add changelog for terra-grid

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Adding changelog for grid since its v1.0.0 release to now.

### Additional Details
We can use this review to decide what format we like for the CHANGELOG file. Based on this review, I'll add a change log to terra-mixins for v1.0.0+ releases and all the other components when we release them as v1.

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
Was this auto generated or manual?

**Comment by bjankord** [comment]
This was manually generated.

File: packages/terra-grid/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+Terra Grid Changelog
+====================
+
+3.1.0 - 2017-06-20
```
Comment (emilyrohrbough) [inline packages/terra-grid/CHANGELOG.md]: Total nit: I would vote towards formatting the date differently, maybe like React's changelog, which would use `(June 20, 2017)`

File: packages/terra-grid/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+------------------
+### Enhancements
```
Comment (emilyrohrbough) [inline packages/terra-grid/CHANGELOG.md]: Would it be helpful consumers to separate dependencies from enhancements by listing dependency changes under it's own section. Then a consumer could quickly see dependency changes verse enhancements in functionality?

Comment (windse7en) [inline packages/terra-grid/CHANGELOG.md]: I'm OK with current CHANGLOG. For dependency, consumers can view the`package.json` directly to understand the required packages for the component. That's also easy to maintain.

Comment (bjankord) [inline packages/terra-grid/CHANGELOG.md]: Yeah I like the idea of these sections:
* Bug Fix
* Breaking Changes
* Removals
* Enhancements

Sometimes the only change between a version are dependency updates. I feel like its good to note that in the changelog when we do that so consumers don't need to diff the code between packages to see what changed.
I'd propose calling this section "Dependency Updates"

**Pull request merged**